### ISSUE TITLE
Undo Redo and Eraser Support

### ISSUE THREAD

**Issue by PawanmSpaceo** [yuppielabel/YPDrawSignatureView#73]
I need an Undo Redo and Eraser Support can we add it?

**Comment by GJNilsen** [comment]
TLDR: No

**Reasoning why**
Since this class is made for signatures, I would think there is no need for an eraser or undo function. There is a “clear signature” method, so that would be sufficient. The reasoning is that you would not erase parts of your signature on paper either. Even if one have multiple first or last names, so the signature consists of separate sections, the event of making a signature is a continuous flow of curves. Try it for yourself to sign your name with a couple of seconds between each letter, and you will see it will not look like your real signature. There is a cadence and flow in it, and its this feature of your signature that is distinctive for it, and something that makes it real.

This class is made for capturing signatures, and not for note taking. I understand some would like to use it for notes, but that is far outside the scope of the class. Also, the class would not cope with the number of strokes when taking notes the way it is built. It becomes sluggish after a while. Undoing would be pretty easy to implement, but erasing would take a lot of work. Each stroke with the eraser had to subtract from the strokes underneath it, splitting existing strokes if they only where partially erased, and adding even more strokes to the array where they are stored. This would affect the sluggishness mentioned above.

If the class was not based on vectors and bitmaps instead, it would be easy to fix. But then again, it would be outside the scope of the class.

To give you some sort of undoing a signature, I suggest you implement the clear button as described in the readme.

**Comment by PawanmSpaceo** [comment]
@GJNilsen Thank you for replying...

**Comment by GJNilsen** [comment]
I hope you understand its not to be a dick about it. There are great note taking classes around already, so I feel it would be almost like a waste of time to implement all the functions needed for being a great note taking class as well. I do understand the ease of having just to add one simple leight weigth source file to a project without the clutter of maybe hundreds of classes you would need for making a great note taking library.

The one-file-one-class approach without any dependencies is something I personally like about this project, and I guess that is the reason why others than myself are using it as well. The class is so simple, that even if I would stop maintaining it, it would take almost no effort to keep it up to date by one self.

**Comment by PawanmSpaceo** [comment]
Yes, I understood actually you are right I'm using this for notes...
